lib/plugins/plugin/plugin.js

```diff
@@ -0,0 +1,298 @@
+      plugin: {
+        usage: 'Plugin management for Serverless',
+        commands: {
+          install: {
+            usage: 'Install and add a plugin to your service',
+            lifecycleEvents: [
+              'install',
+            ],
+            options: {
+              name: {
+                usage: 'The plugin name',
+                required: true,
+                shortcut: 'n',
+              },
+            },
+          },
+          uninstall: {
+            usage: 'Uninstall and remove a plugin from your service',
+            lifecycleEvents: [
+              'uninstall',
+            ],
+            options: {
+              name: {
+                usage: 'The plugin name',
+                required: true,
+                shortcut: 'n',
+              },
+            },
+          },
+          list: {
+            usage: 'Lists all available plugins',
+            lifecycleEvents: [
+              'list',
+            ],
+          },
+          search: {
+            usage: 'Search for plugins',
+            lifecycleEvents: [
+              'search',
+            ],
+            options: {
+              query: {
+                usage: 'Search query',
+                required: true,
+                shortcut: 'q',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'plugin:install:install': () => BbPromise.bind(this).then(this.install),
+      'plugin:uninstall:uninstall': () => BbPromise.bind(this).then(this.uninstall),
+      'plugin:list:list': () => BbPromise.bind(this).then(this.list),
+      'plugin:search:search': () => BbPromise.bind(this).then(this.search),
+    };
+  }
+
+  install() {
+    return BbPromise.bind(this)
+      .then(this.validate)
+      .then(this.getPlugins)
+      .then((plugins) => {
+        const pluginName = this.options.name;
+        const plugin = plugins.find((item) => item.name === pluginName);
+
+        if (plugin) {
+          const servicePath = this.serverless.config.servicePath;
+          const packageJsonFilePath = path.join(servicePath, 'package.json');
+
+          // check if package.json is already present. Otherwise create one
+          if (!fs.existsSync(packageJsonFilePath)) {
+            this.serverless.cli
+              .log('Creating an empty package.json file in your service directory');
+
+            const packageJsonFileContent = {
+              name: this.serverless.service.service,
+              description: '',
+              version: '0.1.0',
+              dependencies: {},
+              devDependencies: {},
+            };
+
+            fse.writeJsonSync(packageJsonFilePath, packageJsonFileContent);
+          }
+
+          // install the package through npm
+          this.serverless.cli
+            .log(`Installing plugin "${pluginName}" (this might take a few seconds...)`);
+
+          childProcess
+            .execSync(`npm install --save-dev ${pluginName}`, {
```

Comment:
Everyone should have NPM installed. 

We can do something like:
1. Try/catch run npm install
2. If fails try/catch yarn add (maybe do a check for yarn like nik mentions before step 1 and use it instead)
3. If fails good old fashioned download plugin tarball